include/onnxruntime/core/framework/allocator.h

```diff
@@ -52,7 +52,7 @@ namespace synchronize {
 class Notification;
 }
 using WaitNotificationFn = std::function<void(Stream&, synchronize::Notification&)>;
-void* AllocateBufferWithOptions(std::shared_ptr<IAllocator>& allocator, size_t size, bool use_reserve, Stream* stream, WaitNotificationFn wait_fn);
+void* AllocateBufferWithOptions(IAllocator* allocator, size_t size, bool use_reserve, Stream* stream, WaitNotificationFn wait_fn);
 
```

Comment:
nit: IAllocator& so the API communicates it requires the value to be non-null (at least the current implementation does).